### DISCUSSION TITLE
Importing large recipe put the tenant into Initializing state

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#13898] [General]
### Describe the bug
I tried to import a large recipe into a tenant. The recipe had a `DataValidation` exception and it was not imported. However, the site went from `Running` to `Initializing`.

The site ended up in `Running` state after a while. 
@jtkech any idea why the tenant status change like this? I am guessing is that the DB was busy importing all the data so the state showed `Initializing` while the database was busy

**Comment by jtkech** [comment]
Yes, when we execute a recipe from the admin UI we set the shell settings to `Initializing` and revert it back to `Running` in a finally block. Note: I already thought about completing this code which is only a little mimic of the setup.

https://github.com/OrchardCMS/OrchardCore/blob/71aef4aea7bda3fd3b0b68a55ad78759fba99550/src/OrchardCore.Modules/OrchardCore.Recipes/Controllers/AdminController.cs#L110-L120

So normally line 119 the tenant is reverted back to `Running`, unless if it get stuck in the recipe execution, or if the `_shellSettings` is no longer the local shell settings instance, which normally only happens if the shell is reloaded.

Do you have some custom recipe steps that may call `UpdateShellSettingsAsync()` that would persist the shell settings as `Initializing` and then that would create a new shell settings instance when reloading the shell?

**Reply by MikeAlhayek** [reply]
I don't think I have custom settings that calls that method. 

Why do we have to pause the site during the import?

**Reply by jtkech** [reply]
Maybe I misunderstood the context, I thought you were executing a recipe from the admin `Configuration > Recipes`, but maybe your recipe is executed while importing through the admin `Import/Export > Package Import`, but in the last case I didn't see any code that would set the shell settings to the `Initializing` state.

> Why do we have to pause the site during the import?

We only do it when running a recipe from  the admin `Configuration > Recipes`, this because during a given time important changes can be done as feature migrations, but this is still possible if multiple instances are running, and all things separately, without executing a recipe, can be done manually and concurrently.

I think the main reason was that before we were able to also run full setup recipes which is no longer allowed, so maybe we could also remove the code that sets temporarily the tenant to `Initializing`.
